### PR TITLE
chore(engine): migrate event sub-process processor

### LINKED ISSUE #4559: Migrate Event-Sub-Process to new BPMN lifecycle processor
**Description**

Refactor the current implementation of the BPMN element to match with the new BPMN lifecycle processors defined in #4333.

### PR DESCRIPTION
**Pull request by saig0**
## Description

* trigger an event sub-process using the event subscription behavior
* remove unused code

## Related issues

closes #4559 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: engine/src/main/java/io/zeebe/engine/nwe/container/SubProcessProcessor.java [hunk 1]
```diff
@@ -57,13 +57,27 @@ public void onActivating(
...
 
-    final var noneStartEvent = element.getNoneStartEvent();
-    if (noneStartEvent == null) {
-      throw new BpmnProcessingException(
-          context, "Expected to activate the none start event of the sub-process but not found.");
-    }
+    if (element.hasNoneStartEvent()) {
+      // embedded sub-process is activated
+      final var noneStartEvent = element.getNoneStartEvent();
+      if (noneStartEvent == null) {
```
Comment (korthout) [inline engine/src/main/java/io/zeebe/engine/nwe/container/SubProcessProcessor.java:63]: can this still be null?

File: engine/src/main/java/io/zeebe/engine/nwe/behavior/BpmnStateTransitionBehavior.java [hunk 2]
```diff
@@ -206,6 +208,8 @@ public void terminateChildInstances(final BpmnElementContext context) {
       }
       stateBehavior.updateElementInstance(elementInstance);
     }
+
+    return activeChildInstances == 0;
```
Comment (korthout) [inline engine/src/main/java/io/zeebe/engine/nwe/behavior/BpmnStateTransitionBehavior.java:212]: I like this change 👍

File: engine/src/main/java/io/zeebe/engine/nwe/behavior/BpmnEventSubscriptionBehavior.java [hunk 3]
```diff
@@ -380,6 +381,56 @@ public void unsubscribeFromEvents(final BpmnElementContext context) {
...
+                  flowScopeInstance.setInterruptingEventKey(eventElementInstanceKey);
+                });
+
+          } else {
+            // activate non-interrupting event sub-process
+            publishActivatingEvent(context, eventElementInstanceKey, record);
+          }
+
+          return eventElementInstanceKey;
+        });
```
Comment (korthout) [inline engine/src/main/java/io/zeebe/engine/nwe/behavior/BpmnEventSubscriptionBehavior.java:431]: The lambda above is very large, can we find a way to make this a little cleaner? Perhaps extract the body of the lambda to a private method or so?

**Review by korthout — COMMENTED** [review]
Looks good, only some minor comments.

File: engine/src/main/java/io/zeebe/engine/nwe/container/SubProcessProcessor.java (hunk 1 above)
Comment (saig0) [reply engine/src/main/java/io/zeebe/engine/nwe/container/SubProcessProcessor.java:63]: It should not. I can remove the `null` checks.

**Comment by saig0** [comment]
@korthout , I applied your hints. Please have a second look.

**Review by korthout — APPROVED** [review]
LGTM 👍

**Pull request merged**